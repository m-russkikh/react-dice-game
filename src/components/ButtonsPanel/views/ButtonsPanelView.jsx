import React from 'react';
import { FormGroup } from 'reactstrap';

import BetButtonControl from '../../BetButtonControl';


function ButtonsPanelView() {
  return (
    <FormGroup row>
      <BetButtonControl betType="high" />
      <BetButtonControl betType="low" />
    </FormGroup>
  );
}

export default ButtonsPanelView;
