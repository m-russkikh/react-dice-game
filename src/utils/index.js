import { DEFAULT_BALANCE } from '../constants/balance';
import { MAX_BET_NUMBER } from '../constants/diceGame';

export function getPreloadedState() {
  let balance = {
    value: DEFAULT_BALANCE,
  };

  try {
    const serializedBalance = localStorage.getItem('balance');

    if (serializedBalance) {
      balance = JSON.parse(serializedBalance);
    }
  } catch (err) {
  //  ignore errors
  }

  return { balance };
}

export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getBetNumber() {
  return Math.floor(Math.random(0, 1) * MAX_BET_NUMBER) + 1;
}
