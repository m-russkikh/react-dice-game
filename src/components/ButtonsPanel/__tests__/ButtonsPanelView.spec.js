import { getShallowWrapper } from '../../../utils/testUtils';
import ButtonsPanelView from '../views/ButtonsPanelView';


describe('ButtonsPanelView', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = getShallowWrapper(ButtonsPanelView);
  });

  describe('render', () => {
    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
