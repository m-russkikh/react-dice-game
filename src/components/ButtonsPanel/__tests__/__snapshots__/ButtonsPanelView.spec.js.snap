// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsPanelView render should display correct 1`] = `
"<FormGroup row={true} tag=\\"div\\">
  <Connect(BetButtonControlContainer) betType=\\"high\\" />
  <Connect(BetButtonControlContainer) betType=\\"low\\" />
</FormGroup>"
`;
